### DISCUSSION TITLE
Leverage openssl automation tests

### DISCUSSION THREAD

**Discussion by songlingatpan** [open-quantum-safe/liboqs#1950] [Ideas]
Regarding automation testing, I think that most future users will interact with liboqs through OpenSSL and oqsprovider. 
Do we plan to leverage existing OpenSSL test cases to cover the end-to-end integration of OpenSSL, oqsprovider, and liboqs? 
I believe the OpenSSL test cases could help identify a number of potential issues within liboqs.

**Comment by SWilson4** [comment]
I believe that OpenSSL uses oqs-provider in some of their own testing already (mostly to test their own provider logic), but I'm not sure to what extent liboqs is actually being tested there. (@baentsch probably knows the answer to that.) I agree that the more we can work together on testing the better it is for everyone.

Are there any particular tests that you have in mind that you would like to try to bring to OQS, @songlingatpan?

**Reply by songlingatpan** [reply]
@SWilson4 
In practice, OpenSSL, oqsprovider, and liboqs are used together to implement PQ TLS. Any issues in the end-to-end OpenSSL + oqsprovider + liboqs solution can affect the usage of liboqs.

OpenSSL is well-suited for systems with memory constraints and performs reliably even when memory reaches its limit. I believe OpenSSL could include test cases that cover scenarios like malloc failures, potential memory leaks, and NULL pointer dereferencing. Leveraging OpenSSL's comprehensive test cases could help uncover issues within liboqs under various conditions.

CC: @dstebila @baentsch
